mixins/provider-mixin.js

```diff
@@ -0,0 +1,38 @@
+export const ProviderMixin = superclass => class extends superclass {
+
+	constructor() {
+		super();
+		this._providers = new Map();
+		this.addEventListener('d2l-request-provider', this._handleRequest);
+	}
+
+	_handleRequest(e) {
```

Comment:
Suggest this gets inlined in the constructor to avoid collisions with other mixins that might have a `_handleRequest` method... and just generally to not pollute the component's namespace unnecessarily.